test/helpers.ts

```diff
@@ -183,7 +186,11 @@ export async function xhr(
         )
       })
     }
-
+    req.timeout = options?.timeout ?? 0
+    req.ontimeout = function (event) {
```

Comment:
I think we shouldn't extend this utility with the timeout functionality, as it introduces too much complexity into this utility function. 

Instead, we should perform a raw `XMLHttpRequest` and assert the proper timeout handling with both callbacks and events attached to XHR instance. I'll push that test suite right now.